specifyweb/stored_queries/execution.py

```diff
@@ -559,7 +559,30 @@ def execute(session, collection, user, tableid, distinct, count_only, field_spec
         if limit:
             query = query.limit(limit)
 
-        return {'results': list(query)}
+        results = list(query)
+
+        age_field_position = None
+        age_value = None
+
+        for index, field_spec in enumerate(field_specs):
+        # Search for the "age" field within the field_spec
+            for field in field_spec.fieldspec.join_path:
+                if field.name == "age":
+                    age_field_position = index
+                    age_value = field_spec.value
+                    break
+
+        # Check if the age field exists and modify the id with age value
+        if age_field_position is not None:
+            results = [
+                tuple(
+                    age_value.replace(",", " - ") if i == age_field_position + 1 else value
+                    for i, value in enumerate(result)
+                )
+                for result in results
+            ]
+
+        return {'results': results}
```

Comment:
I think we should look for a better place to do this if possible. Instead of 'fixing' the query after it has been constructed, it would be better to just construct the query correctly. Looks like the code iterates over each column and each row from the result, which will be super slow for a sufficiently large query. 

Maybe the fix should go wherever the age values are computed? I'd look into this file: https://github.com/specify/specify7/blob/production/specifyweb/specify/geo_time.py